src/components/Dashboard/Capacity/CapacityItem.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import { DonutChart } from 'patternfly-react';
+import PropTypes from 'prop-types';
+
+import { InlineLoading } from '../../Loading';
+
+const pfGetUtilizationDonutTooltipContents = d =>
+  `<span class="donut-tooltip-pf" style="white-space: nowrap;">${Math.round(d[0].ratio * 1000) / 10} ${
+    d[0].name
+  }</span>`;
+
+const LoadingCapacityItem = ({ idSuffix, title, LoadingComponent }) => (
+  <div className="kubevirt-capacity__item">
+    <h2 className="kubevirt-capacity__item-title">{title}</h2>
+    <h6>
+      <LoadingComponent />
+    </h6>
+    <div>
+      <DonutChart
+        id={`donut-chart-${idSuffix}`}
+        size={{ width: 150, height: 150 }}
+        data={{
+          unload: true,
+          columns: [['Used', 0], ['Available', 100]],
+          order: null,
+        }}
+        unloadBeforeLoad
+        title={{
+          primary: 'Loading',
+          secondary: 'Capacity',
+        }}
+      />
+    </div>
+  </div>
+);
+
+LoadingCapacityItem.propTypes = {
+  idSuffix: PropTypes.string.isRequired,
+  title: PropTypes.string.isRequired,
+  LoadingComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
+};
+
+export const CapacityItem = ({ idSuffix, title, used, total, unit, formatValue, LoadingComponent }) => {
+  if (!Number.isNaN(parseFloat(used)) && !Number.isNaN(parseFloat(total))) {
+    const totalFormatted = formatValue(total, unit);
+    const usedFormatted = formatValue(used, totalFormatted.unit);
+
+    unit = totalFormatted.unit; // eslint-disable-line prefer-destructuring
+    total = totalFormatted.value;
+    used = usedFormatted.value;
+
+    const available = total - used;
+    const columns = [['Used', used], ['Available', available]];
+
+    const percentageUsed = total > 0 ? Math.round((100 * used) / total) : 0;
+    const availableNice = Number.isInteger(available) ? available : available.toFixed(1);
+    const totalNice = Number.isInteger(total) ? total : total.toFixed(1);
+
+    return (
+      <div className="kubevirt-capacity__item">
+        <h2 className="kubevirt-capacity__item-title">{title}</h2>
+        <h6>{`${availableNice} ${unit} available out of ${totalNice} ${unit}`}</h6>
+        <div>
+          <DonutChart
+            id={`donut-chart-${idSuffix}`}
```

Comment:
I get that the idSuffix is useful (lazy approach). But I would prefer the id part was configurable as well. Can you either create idPrefix prop with `donut-chart` value as a default or add just an id prop which is more conventional?

Reply:
also there is a function called `prefixedId`

Reply:
prop renamed to `id` and Im using `prefixedId` now